app/ui/src/main/java/com/fsck/k9/ui/folders/FolderIconLoader.kt

```diff
@@ -0,0 +1,60 @@
+package com.fsck.k9.ui.folders
+
+import android.content.res.Resources
+import android.util.TypedValue
+import com.fsck.k9.mail.Folder
+import com.fsck.k9.mailstore.FolderType
+import com.fsck.k9.ui.R
+
+class FolderIconLoader(private val theme: Resources.Theme) {
+    private var iconFolderInboxResId: Int = 0
+    private var iconFolderOutboxResId: Int = 0
+    private var iconFolderSentResId: Int = 0
+    private var iconFolderTrashResId: Int = 0
+    private var iconFolderDraftsResId: Int = 0
+    private var iconFolderArchiveResId: Int = 0
+    private var iconFolderSpamResId: Int = 0
+    public var iconFolderResId: Int = 0
+
+    init {
+        iconFolderInboxResId = getResId(R.attr.iconFolderInbox)
+        iconFolderOutboxResId = getResId(R.attr.iconFolderOutbox)
+        iconFolderSentResId = getResId(R.attr.iconFolderSent)
+        iconFolderTrashResId = getResId(R.attr.iconFolderTrash)
+        iconFolderDraftsResId = getResId(R.attr.iconFolderDrafts)
+        iconFolderArchiveResId = getResId(R.attr.iconFolderArchive)
+        iconFolderSpamResId = getResId(R.attr.iconFolderSpam)
+        iconFolderResId = getResId(R.attr.iconFolder)
+    }
+
+    private fun getResId(resAttribute: Int): Int {
+        val typedValue = TypedValue()
+        val found = theme.resolveAttribute(resAttribute, typedValue, true)
+        if (!found) {
+            throw AssertionError("Couldn't find resource with attribute $resAttribute")
+        }
+        return typedValue.resourceId
+    }
+
+    public fun getFolderIcon(type: FolderType): Int = when (type) {
+        FolderType.INBOX -> iconFolderInboxResId
+        FolderType.OUTBOX -> iconFolderOutboxResId
+        FolderType.SENT -> iconFolderSentResId
+        FolderType.TRASH -> iconFolderTrashResId
+        FolderType.DRAFTS -> iconFolderDraftsResId
+        FolderType.ARCHIVE -> iconFolderArchiveResId
+        FolderType.SPAM -> iconFolderSpamResId
+        else -> iconFolderResId
+    }
+
+    public fun getFolderIcon(type: Folder.FolderType): Int = when (type) {
+        Folder.FolderType.INBOX -> iconFolderInboxResId
```

Comment:
I suggest using an import of `import com.fsck.k9.mail.Folder.FolderType as LegacyFolderType` and replacing all occurrences of `Folder.FolderType` in this file with `LegacyFolderType`. I find this makes the code a little bit easier to read.